src/Webapi/Dom/Webapi__Dom__HtmlOptionsCollection.res

```diff
@@ -0,0 +1,31 @@
+/**
+ * Spec: https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#htmloptionscollection
+ */
+type t = Dom.htmlOptionsCollection
+
+// realistically this only deals with HtmlOption and HtmlOptGroup elements
+
+include Webapi__Dom__HtmlCollection.Impl({
+  type t = t
+})
+
+@set external setLength: (t, int) => unit = "length"
+@set_index external setItem: (t, int, Dom.htmlElement) => unit = ""
+@set_index external clearItem: (t, int, Js.null<'a>) => unit = ""
+
+@get external selectedIndex: t => int = "selectedIndex"
+@set external setSelectedIndex: (t, int) => int = "selectedIndex"
+@set external setSelectedElement: (t, Dom.htmlElement) => Dom.htmlElement = "selectedIndex"
+
+@send external clearSelected: (t, @as(json`-1`) _) => unit = "selectedIndex"
+
+/**
+ * This method will throw a "HierarchyRequestError" DOMException if element is an ancestor of the element into which it is to be inserted.
+ */
+@send
+external add: (
+  t,
+  ~option: Dom.htmlElement,
```

Comment:
This on the other hand could be a unwrapped polymorphic variant for `htmlOptGroupElement` and `htmlOptionElement`.